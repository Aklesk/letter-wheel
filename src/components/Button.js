import React from 'react';
import PropTypes from 'prop-types';
import { ImageBackground, Text, TouchableWithoutFeedback, View } from 'react-native';
import { observer } from 'mobx-react';
import { action, observable } from 'mobx';

import ButtonStyle from '../styles/ButtonStyle';

import buttonYellowPassive from '../images/buttonYellowPassive.png';
import buttonYellowActive from '../images/buttonYellowActive.png';
import buttonBluePassive from '../images/buttonBluePassive.png';
import buttonBlueActive from '../images/buttonBlueActive.png';

@observer
export default class Button extends React.Component {
  static propTypes = {
    colour: PropTypes.oneOf(['yellow', 'blue']),
    title: PropTypes.string,
    onPress: PropTypes.func,
  }
  static defaultProps = {
    colour: 'yellow',
    title: '',
    onPress() {},
  }

  @action beginPress = () => {
    this.isPressing = true;
  }
  @action endPress = () => {
    this.isPressing = false;
  }
  @observable isPressing = false;

  render() {
    let unpressed;
    let pressed;

    if (this.props.colour === 'yellow') {
      unpressed = buttonYellowPassive;
      pressed = buttonYellowActive;
    }
    if (this.props.colour === 'blue') {
      unpressed = buttonBluePassive;
      pressed = buttonBlueActive;
    }

    return (
      <TouchableWithoutFeedback
        onPress={this.props.onPress}
        onPressIn={this.beginPress}
        onPressOut={this.endPress}
        style={ButtonStyle.wrapper}
      >
        <ImageBackground
          fadeDuration={0}
          source={this.isPressing ? pressed : unpressed}
          style={ButtonStyle.image}
        >
          <View pointerEvents="none">
            <Text
              style={ButtonStyle.text}
            >
              {this.props.title.toUpperCase()}
            </Text>
          </View>
        </ImageBackground>
      </TouchableWithoutFeedback>
    );
  }
}
