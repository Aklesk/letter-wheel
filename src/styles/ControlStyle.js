import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    flexDirection: 'column',
    flex: 1,
  },
  timerContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  timerText: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    color: '#000',
    fontSize: 20,
  },
  progressContainer: {
    alignItems: 'center',
    flex: 2,
    justifyContent: 'center',
  },
  progressText: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    color: '#227711',
    fontSize: 30,
    textAlign: 'center',
  },
  progressTextSmall: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    color: '#000',
    fontSize: 16,
    textAlign: 'center',
  },
  resultContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  resultText: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    fontSize: 20,
    color: '#444',
  },
  columnContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  leftColumn: {
    backgroundColor: '#fff',
    borderRadius: 4,
    flex: 4,
    marginBottom: 10,
    marginTop: 10,
    overflow: 'hidden',
  },
  leftColumnHeader: {
    alignItems: 'center',
    backgroundColor: '#b3c9d0',
    justifyContent: 'center',
  },
  leftColumnHeaderText: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    fontSize: 15,
    color: '#444',
    margin: 5,
  },
  rightColumn: {
    flex: 6,
    marginBottom: 10,
    marginLeft: 10,
    marginTop: 10,
  },
  row: {
    marginLeft: 10,
    marginBottom: 10,
  },
  correct: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    color: '#505050',
    fontSize: 20,
  },
  incorrect: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    color: '#a3a3a3',
    fontSize: 20,
    textDecorationLine: 'line-through',
  },
  neutral: {
    fontFamily: 'RujisHandwritingFontv.2.0',
    color: '#444',
    fontSize: 20,
  },
});
