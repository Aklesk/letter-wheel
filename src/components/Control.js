import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  ListView,
  Vibration,
  View,
  Text,
  TouchableOpacity,
} from 'react-native';
import { inject, observer } from 'mobx-react';
import Icon from 'react-native-vector-icons/Ionicons';

import ControlStyle from '../styles/ControlStyle';

@inject('appStore') @observer
export default class Control extends React.Component {
  static propTypes = {
    appStore: PropTypes.shape({
      dataSource: PropTypes.object.isRequired,
      gameModal: PropTypes.object,
      getScore: PropTypes.func.isRequired,
      letters: PropTypes.string.isRequired,
      scores: PropTypes.array.isRequired,
      selected: PropTypes.object.isRequired,
      statusText: PropTypes.string.isRequired,
      submitWord: PropTypes.func.isRequired,
      timer: PropTypes.number.isRequired,
      tried: PropTypes.object.isRequired,
      words: PropTypes.object.isRequired,
    }).isRequired,
    onCorrect: PropTypes.func,
    onWrong: PropTypes.func,
  }
  static defaultProps = {
    onCorrect() {},
    onWrong() {},
  }
  submitWord = () => {
    const wordValidity = this.props.appStore.submitWord();
    if (wordValidity === true) {
      this.props.onCorrect();
    } else if (wordValidity === false) {
      this.props.onWrong();
    }
  }
  render() {
    const { appStore } = this.props;
    const correct = appStore.tried.filter(tryEntry => (
      appStore.words.indexOf(tryEntry.word) !== -1
    )).length;
    const {
      backspaceTouch,
      backspaceWrapper,
      buttonWrapper,
      columnContainer,
      container,
      entryContainer,
      entryWrapper,
      entryText,
      leftColumn,
      leftColumnHeader,
      leftColumnHeaderText,
      resultContainer,
      resultText,
      rightColumn,
      row,
      progressContainer,
      progressText,
      progressTextSmall,
      timerContainer,
      timerText,
    } = ControlStyle;
    const formattedTimer = appStore.timer / 60 >= 1 ?
      `${Math.floor(appStore.timer / 60)}m ${appStore.timer % 60}s` :
      `${appStore.timer % 60}s`;
    const score = appStore.getScore();
    const feedbackText = (() => {
      const { toNext } = score;
      if (toNext > 0) {
        return `${toNext} word${toNext > 1 ? 's' : ''}\n to next level`;
      }
      return '';
    })();
    return (
      <View style={container}>
        <View style={entryContainer}>
          <View style={entryWrapper}>
            <Text style={entryText}>
              {
                appStore.selected.map(i => appStore.letters[i].toUpperCase()).join('')
              }
            </Text>
          </View>
          <View style={backspaceWrapper}>
            <TouchableOpacity
              onPress={() => { appStore.selected.pop(); }}
              onLongPress={() => {
                Vibration.vibrate(100);
                appStore.selected.clear();
              }}
              style={backspaceTouch}
            >
              <Icon
                name="md-backspace"
                size={35}
              />
            </TouchableOpacity>
          </View>
        </View>
        <View style={columnContainer}>

          <View style={leftColumn}>
            <View style={leftColumnHeader}>
              <Text style={leftColumnHeaderText}>
                {`${correct} / ${appStore.words.length}`}
              </Text>
            </View>
            <ListView
              dataSource={appStore.dataSource}
              enableEmptySections
              renderRow={singleRow => (
                <Text style={ControlStyle[singleRow.style]}>
                  {singleRow.word}
                </Text>
              )}
              style={row}
            />
          </View>

          <View style={rightColumn}>
            <View style={resultContainer}>
              <Text style={resultText}>
                {appStore.statusText}
              </Text>
            </View>
            <View style={buttonWrapper}>
              <Button
                color="#999"
                onPress={this.submitWord}
                title="Submit"
              />
            </View>
            {appStore.timer !== -1 ?
              <View style={timerContainer}>
                <Text style={timerText}>
                  {formattedTimer}
                </Text>
              </View>
              :
              <View />
            }
            <View style={progressContainer}>
              <View>
                <Text style={progressText}>
                  {score.text !== '' ? `${score.text}!` : ' '}
                </Text>
              </View>
              <View>
                <Text style={progressTextSmall}>
                  {feedbackText}
                </Text>
              </View>
            </View>
            <View style={buttonWrapper}>
              <Button
                color="#999"
                onPress={() => {
                  appStore.gameModal.close();
                  appStore.gameModal.open();
                }}
                title="Menu"
              />
            </View>
          </View>

        </View>
      </View>
    );
  }
}
