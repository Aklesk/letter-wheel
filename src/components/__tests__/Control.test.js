import React from 'react';
import { ListView } from 'react-native';
import { mount } from 'enzyme';
import { observable } from 'mobx';

import Control from '../Control';
import ControlStyle from '../../styles/ControlStyle';
import Button from '../Button';

describe('Control component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    const render = mount((
      <Control />
    ));
    expect(render).toMatchSnapshot();
    render.setProps({
      isScored: true,
      scoreText: 'Nice!',
      wordsToNextLevel: 5,
      selectedLetters: 'akdc',
      statusText: 'Game scored.',
      timerString: '1m 20s',
      tried: observable.map({
        alit: true,
        yrasd: false,
        built: true,
        bail: true,
        asdf: false,
      }),
      words: observable.map({
        alit: true,
        bail: true,
        built: true,
        hail: true,
        halt: true,
        haul: true,
        tail: true,
      }),
    });
    render.update();
    expect(render).toMatchSnapshot();
  });

  it('provides a formattedTriedWords computed with formatting indications', () => {
    const render = mount((
      <Control />
    ));
    const instance = render.instance();
    expect(instance.formattedTriedWords).toEqual([
      { word: 'No words', style: 'neutral' },
    ]);

    render.props().words.replace({
      alit: true,
      bail: true,
      built: true,
      hail: true,
      halt: true,
      haul: true,
      tail: true,
    });
    expect(instance.formattedTriedWords).toEqual([
      { word: 'No words', style: 'neutral' },
    ]);

    render.props().tried.replace({
      alit: true,
      yrasd: false,
      built: true,
      bail: true,
      asdf: false,
    });
    expect(instance.formattedTriedWords).toEqual([
      { word: 'alit', style: 'correct' },
      { word: 'asdf', style: 'incorrect' },
      { word: 'bail', style: 'correct' },
      { word: 'built', style: 'correct' },
      { word: 'yrasd', style: 'incorrect' },
    ]);

    render.setProps({ isScored: true });
    expect(instance.formattedTriedWords).toEqual([
      { word: 'Not found:', style: 'neutral' },
      { word: 'hail', style: 'neutral' },
      { word: 'halt', style: 'neutral' },
      { word: 'haul', style: 'neutral' },
      { word: 'tail', style: 'neutral' },
      { word: ' ', style: 'neutral' },
      { word: 'Your words:', style: 'neutral' },
      { word: 'alit', style: 'correct' },
      { word: 'asdf', style: 'incorrect' },
      { word: 'bail', style: 'correct' },
      { word: 'built', style: 'correct' },
      { word: 'yrasd', style: 'incorrect' },
    ]);

    render.props().tried.replace({
      alit: true,
      asdf: false,
      bail: true,
      built: true,
      hail: true,
      halt: true,
      haul: true,
      tail: true,
      yrasd: false,
    });
    expect(instance.formattedTriedWords).toEqual([
      { word: ' ', style: 'neutral' },
      { word: 'Your words:', style: 'neutral' },
      { word: 'alit', style: 'correct' },
      { word: 'asdf', style: 'incorrect' },
      { word: 'bail', style: 'correct' },
      { word: 'built', style: 'correct' },
      { word: 'hail', style: 'correct' },
      { word: 'halt', style: 'correct' },
      { word: 'haul', style: 'correct' },
      { word: 'tail', style: 'correct' },
      { word: 'yrasd', style: 'incorrect' },
    ]);

    render.props().tried.clear();
    expect(instance.formattedTriedWords).toEqual([
      { word: 'Not found:', style: 'neutral' },
      { word: 'alit', style: 'neutral' },
      { word: 'bail', style: 'neutral' },
      { word: 'built', style: 'neutral' },
      { word: 'hail', style: 'neutral' },
      { word: 'halt', style: 'neutral' },
      { word: 'haul', style: 'neutral' },
      { word: 'tail', style: 'neutral' },
    ]);
  });

  it('provides a feedbackText computed which returns number of words to level up', () => {
    const render = mount((
      <Control />
    ));
    const instance = render.instance();
    expect(instance.feedbackText).toEqual('');
    render.setProps({ wordsToNextLevel: 1 });
    expect(instance.feedbackText).toEqual('1 word to level up');
    render.setProps({ wordsToNextLevel: 2 });
    expect(instance.feedbackText).toEqual('2 words to level up');
    render.setProps({ wordsToNextLevel: 8 });
    expect(instance.feedbackText).toEqual('8 words to level up');
    render.setProps({ wordsToNextLevel: 0 });
    expect(instance.feedbackText).toEqual('');
  });

  it('provides a submit button', () => {
    const render = mount((
      <Control
        onSubmit={jest.fn()}
      />
    ));
    const submit = render.find(Button).find({ title: 'Submit' }).at(0);
    expect(submit.props().onPress).toEqual(expect.any(Function));
    expect(render.props().onSubmit).toHaveBeenCalledTimes(0);
    submit.props().onPress();
    expect(render.props().onSubmit).toHaveBeenCalledTimes(1);
  });

  it('provides a menu button', () => {
    const render = mount((
      <Control
        onMenu={jest.fn()}
      />
    ));
    const menu = render.find(Button).find({ title: 'Menu' }).at(0);
    expect(menu.props().onPress).toEqual(expect.any(Function));
    expect(render.props().onMenu).toHaveBeenCalledTimes(0);
    menu.props().onPress();
    expect(render.props().onMenu).toHaveBeenCalledTimes(1);
  });

  it('provides a list of tried words', () => {
    const render = mount((
      <Control
        onMenu={jest.fn()}
        isScored
        scoreText="Nice!"
        wordsToNextLevel={5}
        selectedLetters="akdc"
        statusText="Welcome!"
        timerString="30s"
        tried={observable.map({
          bail: true,
          asdf: false,
        })}
        words={observable.map({
          bail: true,
          built: true,
        })}
      />
    ));
    const list = render.find(ListView).at(0);
    // eslint-disable-next-line no-underscore-dangle
    expect(list.props().dataSource._dataBlob).toEqual([
      { word: 'Not found:', style: 'neutral' },
      { word: 'built', style: 'neutral' },
      { word: ' ', style: 'neutral' },
      { word: 'Your words:', style: 'neutral' },
      { word: 'asdf', style: 'incorrect' },
      { word: 'bail', style: 'correct' },
    ]);
    const { renderRow } = list.props();
    const result1 = mount(renderRow({ word: 'bail', style: 'correct' }));
    expect(result1.props().style).toEqual(ControlStyle.correct);
    expect(result1.props().children).toEqual('bail');

    const result2 = mount(renderRow({ word: 'lafe', style: 'incorrect' }));
    expect(result2.props().style).toEqual(ControlStyle.incorrect);
    expect(result2.props().children).toEqual('lafe');

    const result3 = mount(renderRow({ word: 'words', style: 'neutral' }));
    expect(result3.props().style).toEqual(ControlStyle.neutral);
    expect(result3.props().children).toEqual('words');
  });
});
