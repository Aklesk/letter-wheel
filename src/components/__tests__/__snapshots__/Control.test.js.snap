// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control component matches the snapshot 1`] = `
<Control
  isScored={false}
  onMenu={[Function]}
  onSubmit={[Function]}
  scoreText=""
  statusText=""
  timerString=""
  tried={Object {}}
  words={Object {}}
  wordsToNextLevel={0}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <react-native-mock>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <react-native-mock>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "flex": 4,
                "marginBottom": 10,
                "marginTop": 10,
                "overflow": "hidden",
              }
            }
          >
            <react-native-mock>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#b3c9d0",
                    "justifyContent": "center",
                  }
                }
              >
                <react-native-mock>
                  <Text
                    style={
                      Object {
                        "color": "#444",
                        "fontFamily": "RujisHandwritingFontv.2.0",
                        "fontSize": 15,
                        "margin": 5,
                      }
                    }
                  >
                    <react-native-mock>
                      0 / 0
                    </react-native-mock>
                  </Text>
                </react-native-mock>
              </View>
              <ListView
                dataSource={
                  ListViewDataSource {
                    "_dataBlob": Array [
                      Object {
                        "style": "neutral",
                        "word": "No words",
                      },
                    ],
                  }
                }
                enableEmptySections={true}
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                style={
                  Object {
                    "marginBottom": 10,
                    "marginLeft": 10,
                  }
                }
              >
                <react-native-mock />
              </ListView>
            </react-native-mock>
          </View>
          <View
            style={
              Object {
                "flex": 6,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginTop": 10,
              }
            }
          >
            <react-native-mock>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <react-native-mock>
                  <Text
                    style={
                      Object {
                        "color": "#444",
                        "fontFamily": "RujisHandwritingFontv.2.0",
                        "fontSize": 20,
                      }
                    }
                  >
                    <react-native-mock />
                  </Text>
                </react-native-mock>
              </View>
              <View>
                <react-native-mock>
                  <Button
                    colour="blue"
                    onPress={[Function]}
                    title="Submit"
                  >
                    <TouchableWithoutFeedback
                      onPress={[Function]}
                      onPressIn={[Function]}
                      onPressOut={[Function]}
                      style={
                        Object {
                          "height": "auto",
                          "width": 200,
                        }
                      }
                    >
                      <react-native-mock>
                        <ImageBackground
                          source={
                            Object {
                              "testUri": "src/images/buttonBluePassive.png",
                            }
                          }
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "height": undefined,
                              "justifyContent": "center",
                              "paddingBottom": 2,
                              "paddingLeft": 12,
                              "paddingRight": 12,
                              "paddingTop": 8,
                              "width": undefined,
                            }
                          }
                        >
                          <react-native-mock>
                            <View
                              pointerEvents="none"
                            >
                              <react-native-mock>
                                <Text
                                  style={
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "Grundschrift-Bold",
                                      "fontSize": 20,
                                      "marginBottom": 15,
                                    }
                                  }
                                >
                                  <react-native-mock>
                                    SUBMIT
                                  </react-native-mock>
                                </Text>
                              </react-native-mock>
                            </View>
                          </react-native-mock>
                        </ImageBackground>
                      </react-native-mock>
                    </TouchableWithoutFeedback>
                  </Button>
                </react-native-mock>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 2,
                    "justifyContent": "center",
                  }
                }
              >
                <react-native-mock>
                  <View>
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#227711",
                            "fontFamily": "RujisHandwritingFontv.2.0",
                            "fontSize": 30,
                            "textAlign": "center",
                          }
                        }
                      >
                        <react-native-mock>
                           
                        </react-native-mock>
                      </Text>
                    </react-native-mock>
                  </View>
                  <View>
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontFamily": "RujisHandwritingFontv.2.0",
                            "fontSize": 16,
                            "textAlign": "center",
                          }
                        }
                      >
                        <react-native-mock />
                      </Text>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
              <View>
                <react-native-mock>
                  <Button
                    colour="yellow"
                    onPress={[Function]}
                    title="Menu"
                  >
                    <TouchableWithoutFeedback
                      onPress={[Function]}
                      onPressIn={[Function]}
                      onPressOut={[Function]}
                      style={
                        Object {
                          "height": "auto",
                          "width": 200,
                        }
                      }
                    >
                      <react-native-mock>
                        <ImageBackground
                          source={
                            Object {
                              "testUri": "src/images/buttonYellowPassive.png",
                            }
                          }
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "height": undefined,
                              "justifyContent": "center",
                              "paddingBottom": 2,
                              "paddingLeft": 12,
                              "paddingRight": 12,
                              "paddingTop": 8,
                              "width": undefined,
                            }
                          }
                        >
                          <react-native-mock>
                            <View
                              pointerEvents="none"
                            >
                              <react-native-mock>
                                <Text
                                  style={
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "Grundschrift-Bold",
                                      "fontSize": 20,
                                      "marginBottom": 15,
                                    }
                                  }
                                >
                                  <react-native-mock>
                                    MENU
                                  </react-native-mock>
                                </Text>
                              </react-native-mock>
                            </View>
                          </react-native-mock>
                        </ImageBackground>
                      </react-native-mock>
                    </TouchableWithoutFeedback>
                  </Button>
                </react-native-mock>
              </View>
            </react-native-mock>
          </View>
        </react-native-mock>
      </View>
    </react-native-mock>
  </View>
</Control>
`;

exports[`Control component matches the snapshot 2`] = `
<Control
  isScored={true}
  onMenu={[Function]}
  onSubmit={[Function]}
  scoreText="Nice!"
  selectedLetters="akdc"
  statusText="Game scored."
  timerString="1m 20s"
  tried={
    Object {
      "alit": true,
      "asdf": false,
      "bail": true,
      "built": true,
      "yrasd": false,
    }
  }
  words={
    Object {
      "alit": true,
      "bail": true,
      "built": true,
      "hail": true,
      "halt": true,
      "haul": true,
      "tail": true,
    }
  }
  wordsToNextLevel={5}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <react-native-mock>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <react-native-mock>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "flex": 4,
                "marginBottom": 10,
                "marginTop": 10,
                "overflow": "hidden",
              }
            }
          >
            <react-native-mock>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#b3c9d0",
                    "justifyContent": "center",
                  }
                }
              >
                <react-native-mock>
                  <Text
                    style={
                      Object {
                        "color": "#444",
                        "fontFamily": "RujisHandwritingFontv.2.0",
                        "fontSize": 15,
                        "margin": 5,
                      }
                    }
                  >
                    <react-native-mock>
                      3 / 7
                    </react-native-mock>
                  </Text>
                </react-native-mock>
              </View>
              <ListView
                dataSource={
                  ListViewDataSource {
                    "_dataBlob": Array [
                      Object {
                        "style": "neutral",
                        "word": "Not found:",
                      },
                      Object {
                        "style": "neutral",
                        "word": "hail",
                      },
                      Object {
                        "style": "neutral",
                        "word": "halt",
                      },
                      Object {
                        "style": "neutral",
                        "word": "haul",
                      },
                      Object {
                        "style": "neutral",
                        "word": "tail",
                      },
                      Object {
                        "style": "neutral",
                        "word": " ",
                      },
                      Object {
                        "style": "neutral",
                        "word": "Your words:",
                      },
                      Object {
                        "style": "correct",
                        "word": "alit",
                      },
                      Object {
                        "style": "incorrect",
                        "word": "asdf",
                      },
                      Object {
                        "style": "correct",
                        "word": "bail",
                      },
                      Object {
                        "style": "correct",
                        "word": "built",
                      },
                      Object {
                        "style": "incorrect",
                        "word": "yrasd",
                      },
                    ],
                  }
                }
                enableEmptySections={true}
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                style={
                  Object {
                    "marginBottom": 10,
                    "marginLeft": 10,
                  }
                }
              >
                <react-native-mock />
              </ListView>
            </react-native-mock>
          </View>
          <View
            style={
              Object {
                "flex": 6,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginTop": 10,
              }
            }
          >
            <react-native-mock>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <react-native-mock>
                  <Text
                    style={
                      Object {
                        "color": "#444",
                        "fontFamily": "RujisHandwritingFontv.2.0",
                        "fontSize": 20,
                      }
                    }
                  >
                    <react-native-mock>
                      Game scored.
                    </react-native-mock>
                  </Text>
                </react-native-mock>
              </View>
              <View>
                <react-native-mock>
                  <Button
                    colour="blue"
                    onPress={[Function]}
                    title="Submit"
                  >
                    <TouchableWithoutFeedback
                      onPress={[Function]}
                      onPressIn={[Function]}
                      onPressOut={[Function]}
                      style={
                        Object {
                          "height": "auto",
                          "width": 200,
                        }
                      }
                    >
                      <react-native-mock>
                        <ImageBackground
                          source={
                            Object {
                              "testUri": "src/images/buttonBluePassive.png",
                            }
                          }
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "height": undefined,
                              "justifyContent": "center",
                              "paddingBottom": 2,
                              "paddingLeft": 12,
                              "paddingRight": 12,
                              "paddingTop": 8,
                              "width": undefined,
                            }
                          }
                        >
                          <react-native-mock>
                            <View
                              pointerEvents="none"
                            >
                              <react-native-mock>
                                <Text
                                  style={
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "Grundschrift-Bold",
                                      "fontSize": 20,
                                      "marginBottom": 15,
                                    }
                                  }
                                >
                                  <react-native-mock>
                                    SUBMIT
                                  </react-native-mock>
                                </Text>
                              </react-native-mock>
                            </View>
                          </react-native-mock>
                        </ImageBackground>
                      </react-native-mock>
                    </TouchableWithoutFeedback>
                  </Button>
                </react-native-mock>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <react-native-mock>
                  <Text
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "RujisHandwritingFontv.2.0",
                        "fontSize": 20,
                      }
                    }
                  >
                    <react-native-mock>
                      1m 20s
                    </react-native-mock>
                  </Text>
                </react-native-mock>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 2,
                    "justifyContent": "center",
                  }
                }
              >
                <react-native-mock>
                  <View>
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#227711",
                            "fontFamily": "RujisHandwritingFontv.2.0",
                            "fontSize": 30,
                            "textAlign": "center",
                          }
                        }
                      >
                        <react-native-mock>
                          Nice!
                        </react-native-mock>
                      </Text>
                    </react-native-mock>
                  </View>
                  <View>
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontFamily": "RujisHandwritingFontv.2.0",
                            "fontSize": 16,
                            "textAlign": "center",
                          }
                        }
                      >
                        <react-native-mock>
                          5 words to level up
                        </react-native-mock>
                      </Text>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
              <View>
                <react-native-mock>
                  <Button
                    colour="yellow"
                    onPress={[Function]}
                    title="Menu"
                  >
                    <TouchableWithoutFeedback
                      onPress={[Function]}
                      onPressIn={[Function]}
                      onPressOut={[Function]}
                      style={
                        Object {
                          "height": "auto",
                          "width": 200,
                        }
                      }
                    >
                      <react-native-mock>
                        <ImageBackground
                          source={
                            Object {
                              "testUri": "src/images/buttonYellowPassive.png",
                            }
                          }
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "height": undefined,
                              "justifyContent": "center",
                              "paddingBottom": 2,
                              "paddingLeft": 12,
                              "paddingRight": 12,
                              "paddingTop": 8,
                              "width": undefined,
                            }
                          }
                        >
                          <react-native-mock>
                            <View
                              pointerEvents="none"
                            >
                              <react-native-mock>
                                <Text
                                  style={
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "Grundschrift-Bold",
                                      "fontSize": 20,
                                      "marginBottom": 15,
                                    }
                                  }
                                >
                                  <react-native-mock>
                                    MENU
                                  </react-native-mock>
                                </Text>
                              </react-native-mock>
                            </View>
                          </react-native-mock>
                        </ImageBackground>
                      </react-native-mock>
                    </TouchableWithoutFeedback>
                  </Button>
                </react-native-mock>
              </View>
            </react-native-mock>
          </View>
        </react-native-mock>
      </View>
    </react-native-mock>
  </View>
</Control>
`;
